### PR TITLE
Safer configuration

### PR DESCRIPTION
**Pull request by ikapelyukhin**
1. Added default values for database section;
2. Now uses safe navigation when accessing nested parameters.

### CODE REVIEW

File: lib/rmt/http_request.rb [hunk 1]
```diff
@@ -5,14 +5,14 @@ class RMT::HttpRequest < Typhoeus::Request
 
   def set_defaults
     Typhoeus::Config.user_agent = "RMT/#{RMT::VERSION}"
-    Typhoeus::Config.verbose = Settings.http_client.verbose || nil
+    Typhoeus::Config.verbose = Settings.try(:http_client).try(:verbose) || nil
```
Comment (thutterer) [inline lib/rmt/http_request.rb]: The ` || nil` at the end should not be required anymore, right?

Comment (ikapelyukhin) [reply lib/rmt/http_request.rb]: Right!

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Unless this is a trivial change, please include a CHANGELOG entry.<br />
Run <code>osc vc</code> in the <code>package</code> directory to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by lagartoflojo** [comment]
As you may remember, I'm not a fan of `try` (nor ruby's safe operator `.&`)..
It basically makes code insecure about itself... is the thing there? or is it not? :scream: if it's not, you have to check for `nil`s all over the place. => Sadness.

How would I prefer to solve it in this case? We could:

- Never use the `Settings` class directly, only from `lib/rmt/config.rb`. Why do we have some settings accessed from `Settings` and yet others from `RMT::Config`?
- [Prepend a default config](https://github.com/railsconfig/config#adding-sources-at-runtime) to the `config` gem, which is included somewhere in the RMT sources, ie. it's not user-modifiable.

I'm leaving this here as a comment to open the discussion, not as a ship-stopping review :)

File: config/boot.rb [hunk 2]
```diff
@@ -7,4 +7,4 @@
 require 'rails/commands/server/server_command'
 
 Rails::Command::ServerCommand.send(:remove_const, 'DEFAULT_PORT')
-Rails::Command::ServerCommand.const_set('DEFAULT_PORT', Settings&.[](:rails)&.[](:port) || 4224)
```
Comment (ikapelyukhin) [inline config/boot.rb]: `Settings&.[](:rails)&.[](:port)` -- so pretty :grimacing:

**Comment by ikapelyukhin** [comment]
@lagartoflojo Yeah, the config needs some love. It would be actually cool to use schema validation for the config, it is supported by `railsconfig` gem in some form.

I suggest we add the task to the backlog and properly plan it later.

**Comment by lagartoflojo** [comment]
> It would be actually cool to use schema validation for the config,

Oh yeah, that's cool! https://github.com/railsconfig/config#validation

**Comment by tmuntaner** [comment]
But I agree that it's worth it to clean up the config properly in the future. It can be a good candidate for the maintenance column.

**Pull request merged**